src/detect-http-hh.c

```diff
@@ -126,30 +126,41 @@ static void DetectHttpHostSetupCallback(Signature *s)
     s->mask |= SIG_MASK_REQUIRE_HTTP_STATE;
 }
 
-static _Bool DetectHttpHostValidateCallback(const Signature *s)
+static _Bool DetectHttpHostValidateCallback(const Signature *s, char **sigerror)
 {
     const SigMatch *sm = s->init_data->smlists[g_http_host_buffer_id];
     for ( ; sm != NULL; sm = sm->next) {
         if (sm->type == DETECT_CONTENT) {
             DetectContentData *cd = (DetectContentData *)sm->ctx;
             if (cd->flags & DETECT_CONTENT_NOCASE) {
-                SCLogWarning(SC_ERR_INVALID_SIGNATURE, "http_host keyword "
+                *sigerror = SCStrdup("http_host keyword "
```

Comment:
SCStrdup'ing this shouldn't be necessary. All 'sigerrors' are static strings so we can do without the dup and free later.